src/probnum/diffeq/odefiltsmooth/initialise.py

```diff
@@ -0,0 +1,134 @@
+"""Initialisation procedures."""
+# pylint: disable=import-outside-toplevel
+
+
+from probnum.problems import InitialValueProblem
+
+__all__ = ["compute_all_derivatives"]
+
+# pylint: disable=import-outside-toplevel
+
+
+def compute_all_derivatives(ivp, order=6):
+    r"""Compute derivatives of initial values of an initial value ODE problem.
+
+    This requires jax. For an explanation of what happens "under the hood", see [1]_.
+
+
+    Parameters
+    ----------
+    ivp: InitialValueProblem
+        Initial value problem. See `probnum.problems`.
+    order : int
+        How many derivatives are required? Optional, default is 6. This will later be referred to as :math:`\nu`.
+
+    Raises
+    ------
+    ImportError
+        If `jax` or `jaxlib` are not available.
+
+    Returns
+    -------
+    InitialValueProblem
+        InitialValueProblem object where the attribute `dy0_all` is filled with an
+        :math:`d(\nu + 1)` vector, where :math:`\nu` is the specified order,
+        and :math:`d` is the dimension of the IVP.
+
+    References
+    ----------
+    .. [1] Krämer, N. and Hennig, P., Stable implementation of probabilistic ODE solvers,
+       *arXiv:2012.10106*, 2020.
+
+
+    Examples
+    --------
+    >>> from probnum.problems.zoo.diffeq import threebody_jax, vanderpol_jax
+
+    Compute the initial values of the restricted three-body problem as follows
+
+    >>> res2bod = threebody_jax()
+    >>> print(res2bod.y0)
+    [ 0.994       0.          0.         -2.00158511]
+    >>> print(res2bod.dy0_all)
+    None
+    >>> res2bod = compute_all_derivatives(res2bod, order=3)
+    >>> print(res2bod.y0)
+    [ 0.994       0.          0.         -2.00158511]
+    >>> print(res2bod.dy0_all)
+    [ 9.94000000e-01  0.00000000e+00  0.00000000e+00 -2.00158511e+00
+      0.00000000e+00 -2.00158511e+00 -3.15543023e+02  0.00000000e+00
+     -3.15543023e+02  0.00000000e+00  0.00000000e+00  9.99720945e+04
+      0.00000000e+00  9.99720945e+04  6.39028111e+07  0.00000000e+00]
+
+    Compute the initial values of the van-der-Pol oscillator as follows
+
+    >>> vdp = vanderpol_jax()
+    >>> print(vdp.y0)
+    [2. 0.]
+    >>> print(vdp.dy0_all)
+    None
+    >>> vdp = compute_all_derivatives(vdp, order=3)
+    >>> print(vdp.y0)
+    [2. 0.]
+    >>> print(vdp.dy0_all)
+    [    2.     0.     0.    -2.    -2.    60.    60. -1798.]
+    """
+    all_initial_derivatives = _taylormode(f=ivp.f, z0=ivp.y0, t0=ivp.t0, order=order)
+    return InitialValueProblem(
+        f=ivp.f,
+        t0=ivp.t0,
+        tmax=ivp.tmax,
+        y0=ivp.y0,
+        df=ivp.df,
+        ddf=ivp.ddf,
+        solution=ivp.solution,
+        dy0_all=all_initial_derivatives,
+    )
+
+
+def _taylormode(f, z0, t0, order):
```

Comment:
this function is "hidden", because it is only used and IMO should not be part of the API (we are not offering taylormode, but just use it for initial values. 

Also, I am aware that the code in this function is fairly messy. TBH I am not 100 percent sure how it does what it does, (a source is provided), and I expect it to be cleaned up in the future. FOr now, I just want to get this initialisation thing started ASAP to make some real progress in diffeq.